platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r813/JvmArgumentPassingCrossVersionTest.groovy

```diff
@@ -16,15 +16,19 @@
 
 package org.gradle.integtests.tooling.r813
 
-import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.fixtures.jvm.JDWPUtil
 import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
 import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.test.fixtures.ConcurrentTestUtil
 import org.gradle.test.precondition.Requires
 import org.gradle.test.preconditions.IntegTestPreconditions
 import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.ResultHandler
+import org.junit.Assume
+import spock.lang.IgnoreRest
+import spock.lang.Issue
 
 @ToolingApiVersion(">=8.13")
-@TargetGradleVersion(">=8.13")
```

Comment:
I think these tests are missing a few other interactions:
- What happens with other types of JVM args (not just system properties)
- What happens when the build launcher also has system properties set. how are these combined with jvm args?

Reply:
I've added coverage for the first missing scenario. I'll look into the second later.

Reply:
I've added test coverage to the other other scenario as well. Unfortunately, the existing behaviour is kind of broken: system properties defined in `gradle.properties` files take precedence over the ones defined in the TAPI client configuration. I've documented that in the test. At least now we have evidence that we did not change the behaviour.